### ISSUE TITLE
Does OpenMM compile with swig 3.0.3

### ISSUE THREAD

**Issue by rmcgibbo** [pandegroup/openmm#770]
On windows, I started with swig 3.0.3, and got the following failure:

```
Build FAILED.

"C:\Users\Administrator\openmm\build\ALL_BUILD.vcxproj" (default target) (1) ->
"C:\Users\Administrator\openmm\build\wrappers\python\BuildModule.vcxproj" (default target) (2) ->
(CustomBuild target) ->
  CUSTOMBUILD : warning : Tag `SYMBOL_CACHE_SIZE' at line 289 of file `Doxyfile' has become obsolete. [C:\Users\Adminis
trator\openmm\build\wrappers\python\BuildModule.vcxproj]
  CUSTOMBUILD : warning : Tag `SHOW_DIRECTORIES' at line 473 of file `Doxyfile' has become obsolete. [C:\Users\Administ
rator\openmm\build\wrappers\python\BuildModule.vcxproj]
  CUSTOMBUILD : warning : Tag `HTML_ALIGN_MEMBERS' at line 816 of file `Doxyfile' has become obsolete. [C:\Users\Admini
strator\openmm\build\wrappers\python\BuildModule.vcxproj]
  CUSTOMBUILD : warning : Tag `XML_SCHEMA' at line 1157 of file `Doxyfile' has become obsolete. [C:\Users\Administrator
\openmm\build\wrappers\python\BuildModule.vcxproj]
  CUSTOMBUILD : warning : Tag `XML_DTD' at line 1163 of file `Doxyfile' has become obsolete. [C:\Users\Administrator\op
enmm\build\wrappers\python\BuildModule.vcxproj]
  CUSTOMBUILD : warning : doxygen no longer ships with the FreeSans font. [C:\Users\Administrator\openmm\build\wrappers
\python\BuildModule.vcxproj]


"C:\Users\Administrator\openmm\build\ALL_BUILD.vcxproj" (default target) (1) ->
"C:\Users\Administrator\openmm\build\wrappers\python\BuildModule.vcxproj" (default target) (2) ->
(CustomBuild target) ->
  .\swig_lib\python\extend.i(420): error : Unknown SWIG preprocessor directive: Look [C:\Users\Administrator\openmm\bui
ld\wrappers\python\BuildModule.vcxproj]

    6 Warning(s)
    1 Error(s)

Time Elapsed 00:00:23.22
PS C:\Users\Administrator\openmm\build>
```

Switching to swig 2.0.11 seemed to fix the issue (the build succeeded). Is this expected?

**Comment by rmcgibbo** [comment]
FWIW, I'm compiling with `mkdir build; cd build; cmake ..; cmake --build .`

**Comment by kyleabeauchamp** [comment]
Looks like there's a conda package for linux if we want to try 3.0.2:
http://repo.continuum.io/pkgs/free/linux-64/swig-3.0.2-0.tar.bz2

**Comment by jchodera** [comment]
I was under the impression `swig >= 3.0` was fine, but I realize that the manual does not specify which versions of `swig` are needed under "Prerequisites":
https://github.com/pandegroup/openmm/blob/master/docs-source/usersguide/library.rst

**Comment by peastman** [comment]
I've always used SWIG 2.  I hadn't realized 3 was out.  Ummm... the line it's complaining about is a comment, properly prefixed with #, in the middle of a %pythoncode block.  I don't know why it would have any problem with that.

**Comment by mboisson** [comment]
I to have the same problem. I am trying to compile OpenMM from the sources because the binaries do not work on our system (they assumed a specific path for nvcc, which is not right for our system). I am using swig 3.0.3.

**Comment by rmcgibbo** [comment]
@mboisson: What OS are you on?

**Comment by mboisson** [comment]
Linux CentOS 6.5.

**Comment by swails** [comment]
> I to have the same problem. I am trying to compile OpenMM from the sources because the binaries do not work on our system (they assumed a specific path for nvcc, which is not right for our system). I am using swig 3.0.3.

OpenMM doesn't assume a particular location for `nvcc`; but it _does_ assume a particular version (I think NVCC 6.5 for the latest version).

If your CUDA compiler is not in `/usr/local/cuda/bin`, you can set the `OPENMM_CUDA_COMPILER` environment variable to point to where it *actually* exists.  Note, you'd have to do this even if you compiled from source:

```bash
export OPENMM_CUDA_COMPILER="/path/to/wherever/you/have/nvcc"
```

Of course this doesn't fix SWIG 3.0 with OpenMM (I can try building SWIG 3 on my machine to see if I get the same behavior), but it will at least let you use the CUDA platform with the pre-built binaries.

**Comment by rmcgibbo** [comment]
Swig 3.02 definitely works in *nix, but I haven't tried swig 3.03 again since opening this issue. I've been using 3.02 on *nix and 2.0.11 on windows.

**Comment by mboisson** [comment]
Good to know. The error was only saying that /usr/local/cuda/bin/nvcc could not be used. You may want to include that OPENMM_CUDA_COMPILER info in the error message.

**Comment by jchodera** [comment]
> Good to know. The error was only saying that /usr/local/cuda/bin/nvcc could not be used. You may want to include that OPENMM_CUDA_COMPILER info in the error message.

Sounds like a great idea!

@rmcgibbo : I wonder if there's some way to have a conda post-link script search some of the "usual suspects" for `nvcc`, and set this---or at least warn if it needs to be set.

**Comment by mboisson** [comment]
I guess the discussion is deviating from the main topic here, but why isn't $(which nvcc) used for nvcc instead of a fixed path ?

**Comment by peastman** [comment]
Because nvcc usually isn't in your path.

**Comment by swails** [comment]
This is especially true on clusters where compute nodes often behave differently than head nodes

**Comment by mboisson** [comment]
All the clusters I have used use an environment module system, there is typically a cuda module which set the paths to whatever it should be on that particular cluster.

**Comment by rmcgibbo** [comment]
Probably closed by #875
